Packs/CommonScripts/Scripts/ReadPDFFileV2/ReadPDFFileV2_test.py

```diff
@@ -177,7 +136,137 @@ def test_build_readpdf_entry_object_empty_extract(mocker):
+    if os.path.exists(dec_file_path):
+        os.remove(dec_file_path)
+
+    assert len(urls) == len(expected_output)
+    assert urls == expected_output
+
+
+@pytest.mark.parametrize('file_path', [
+    'URLs_Extraction_Test_PDF_Encoding_Google_Docs_Renderer.pdf',
+    'URLs_Extraction_Test_PDF_Encoding_Quartz_PDFContext.pdf'
+])
+def test_get_urls_and_emails_from_pdf_file_without_encrypt(file_path):
+    """
+    This test verifies URL and Emails extraction from a non-encrypted PDF file.
+
+        Given:
+        A path to a PDF file with a certain encoding:
+            1. A pdf created with google docs.
+            2. A pdf created with mac os Notes.
+
+        Both PDFs include URLs and Email addresses from different kinds that should be extracted:
+
+            * 'https://test1.com/' - A text url ended with a slash /.
+            * 'https://test2.com' - A text url ended without a slash /.
+            * 'www.test3.net' - A text url without the http prefix.
+            * 'user@test4.com' - A text email address.
+            * 'https://test5.com.co/ed/trn/update?email=user@test6.net' - A text url with an https prefix, and an email
+               address in it.
+            * 'http://www.test7.com' - A text hyperlink of a url.
+            * 'https://test8.com/' - An embedded url (a url that is hyperlinked to an image).
+
+        When:
+            Running 'get_urls_and_emails_from_pdf_file' function on the PDF file.
+
+        Then:
+            Verify that the expected amount of URLs and Email addresses was extracted from the PDF.
+            Verify that the URLs Emails was extracted successfully.
+
+    """
+    from ReadPDFFileV2 import get_urls_and_emails_from_pdf_file
+
+    expected_output = {'https://test1.com/', 'https://test2.com', 'http://www.test3.net', 'mailto:user@test4.com',
+                       'https://test5.com.co/ed/trn/update?email=user@test6.net', 'http://www.test7.com',
+                       'https://test8.com/'}
+
+    file_path = f'{CWD}/{file_path}'
+
+    # Extract URLs and Emails:
+    urls = get_urls_and_emails_from_pdf_file(file_path)
+
+    assert len(urls) == len(expected_output)
+    assert urls == expected_output
+
+
+def test_separate_urls_and_emails():
+    """
+        Given:
+        A set including urls and emails that were extracted from a PDF file.
+
+        When:
+            Running 'separate_urls_and_emails' function on the given set.
+
+        Then:
+            Verify the expected amount of URLs identified.
+            Verify the expected amount of Emails identified.
+            Verify that each url was classified as a url, ans email address was classified as an email.
+            Verify that the 'special url' that included both url and email was classified as a url, and the inner email
+            was classified as an email.
+    """
+    from ReadPDFFileV2 import separate_urls_and_emails
+
+    urls_and_emails_input_set = {'https://test.com/', 'www.test.net',
```

Comment:
Not really good practice to hardcode copied values. Lets make this set by having it be the union of expected_urls and expected_emails

Reply:
But it isn't equal to their union.